### ISSUE TITLE
Variable 'F10' undeclared

### ISSUE THREAD

**Issue by ctjacobs** [coneoproject/COFFEE#16]
I'm seeing another undeclared variable error when trying to run the swe_dam_break_2d problem in Firedrake-Fluids. I will try to come up with a minimal failing example, but in the meantime, here's the .c and .err files:

https://gist.github.com/ctjacobs/3edc2d806a135283e89d
https://gist.github.com/ctjacobs/f7ffc70cfd0785b04a52

**Comment by ctjacobs** [comment]
So this code fails with `C = 1.0/tanh(g) - 1.0/g` (which is what Firedrake-Fluids is using), but not with `C = 1.0/tanh(g)`:

```
from firedrake import *

mesh = UnitSquareMesh(2, 2)
H = FunctionSpace(mesh, "CG", 1)

solution = Function(H)
u = TrialFunction(H)
w = TestFunction(H)

g = Function(H).interpolate(Expression("1.0"))
C = 1.0/tanh(g) - 1.0/g
F = C*inner(w, u)*dx

solve(lhs(F) == rhs(F), solution)
```

**Comment by wence-** [comment]
So I think the problem here is that `tanh(g)`, and indeed other math functions, is treated as a `Symbol` in the coffee AST, rather than being a function call with arguments.  I suspect that the necessary change is in FFC to make `_math_function` return a fully-fledged `FunCall("tanh", Symbol(F0))`` rather than just `Symbol("tanh(F0)")` or whatever.,

**Comment by FabioLuporini** [comment]
PR #17  together with PR 65 in FFC should fix the issue

**Comment by ctjacobs** [comment]
This fixes the small example code above, but my swe_dam_break_2d simulation is now failing with:

https://gist.github.com/ctjacobs/48222ac3e3064c2487cb
https://gist.github.com/ctjacobs/7098a625bcc81809eea0

**Comment by FabioLuporini** [comment]
Could you write a minimal example for that?

**Comment by ctjacobs** [comment]
```
from firedrake import *

mesh = UnitSquareMesh(2, 2)
U = VectorFunctionSpace(mesh, "CG", 1)
H = FunctionSpace(mesh, "CG", 1)

u = Function(U).interpolate(Expression(("1.0", "1.0")))
magnitude = TrialFunction(H)
w = TestFunction(H)
solution = Function(H)

a = w*magnitude*dx
L = w*sqrt(dot(u, u))*dx
solve(a == L, solution, bcs=[])
```

**Comment by FabioLuporini** [comment]
Ah, sure, my mistake. Can you update *FFC* (same branch: fix-funcall-gen) and try now? thanks!

**Comment by ctjacobs** [comment]
Thanks - this works now.

**Comment by FabioLuporini** [comment]
Ok, thank you. I've merged #17, but before closing the issue I prefer to wait for the FFC merge

**Comment by FabioLuporini** [comment]
Closing since fixed by PR #21
